controls/V-72315.rb

```diff
@@ -64,30 +106,64 @@
   forward-ports:
   icmp-blocks:
   rich rules:
- rule family=\"ipv4\" source address=\"92.188.21.1/24\" accept
- rule family=\"ipv4\" source address=\"211.17.142.46/32\" accept
-
+  rule family=\"ipv4\" source address=\"92.188.21.1/24\" accept
+  rule family=\"ipv4\" source address=\"211.17.142.46/32\" accept
 If \"firewalld\" is not active, determine whether \"tcpwrappers\" is being used by
 checking whether the \"hosts.allow\" and \"hosts.deny\" files are empty with the
 following commands:
-
 # ls -al /etc/hosts.allow
 rw-r----- 1 root root 9 Aug  2 23:13 /etc/hosts.allow
-
 # ls -al /etc/hosts.deny
 -rw-r----- 1 root root  9 Apr  9  2007 /etc/hosts.deny
-
 If \"firewalld\" and \"tcpwrappers\" are not installed, configured, and active, ask
 the SA if another access control program (such as iptables) is installed and active.
 Ask the SA to show that the running configuration grants or denies access to
 specific hosts or services.
-
 If \"firewalld\" is active and is not configured to grant access to specific hosts
 and \"tcpwrappers\" is not configured to grant or deny access to specific hosts,
 this is a finding."
   tag "fix": "If \"firewalld\" is installed and active on the system, configure
 rules for allowing specific services and hosts.
-
 If \"tcpwrappers\" is installed, configure the \"/etc/hosts.allow\" and
 \"/etc/hosts.deny\" to allow or deny access to specific hosts."
+
+  if service('firewalld').running?
+    @default_zone = firewalld.default_zone
+
+    describe firewalld.where{ zone = @default_zone } do
+      its('services') { should be_in FIREWALLD_SERVICES }
+    end
+
+    describe firewalld do
+      FIREWALLD_HOSTS_ALLOW.each do |rule|
+        it { should have_rule_enabled(rule) }
+      end
+      FIREWALLD_HOSTS_DENY.each do |rule|
+        it { should_not have_rule_enabled(rule) }
+      end
+      FIREWALLD_PORTS_ALLOW.each do |port|
+        it { should have_port_enabled_in_zone(port) }
+      end
+      FIREWALLD_PORTS_DENY.each do |port|
+        it { should_not have_port_enabled_in_zone(port) }
+      end
+    end
+
+  else
+    describe package('tcp_wrappers') do
+      it { should be_installed }
+    end
+    TCPWRAPPERS_ALLOW.each do |rule|
+      describe etc_hosts_allow.where { daemon == rule['daemon'] } do
+        its('client_list') { should include rule['client_list'] }
```

Comment:
Haven't checked but did you validate that the rules handle `DDQ` notation properly? `tcpwrappers` cannot handle `CIDR`.